### PR TITLE
Add basic docs for HTTP API

### PR DESCRIPTION
**Pull request by lloydmeta**
Added some more to the HTTP API page. TODO: Example code.

### CODE REVIEW

File: source/http-api.html.erb [hunk 1]
```diff
@@ -3,4 +3,74 @@ title: HTTP API
...
+    The REST API of Octoparts is relatively small. Those already familiar with <%= link_to "Swagger", "http://swagger.wordnik.com/" %> can simply head to the built-in Swagger documentation, which is hosted at <%= link_to "http://octoparts.herokuapp.com/swagger-ui", "octoparts.herokuapp.com/swagger-ui" %>(not up as of writing).
+  </p>
+</section>
+
+<section>
+  <section>
+    <h2>Basic overview</h2>
+
+    <p>
+      The API consists of a registered dependencies discovery endpoint, a single endpoint for invoking registered dependencies, and a set of cache invalidation endpoints.
```
Comment (cb372) [inline source/http-api.html.erb]: Maybe use bullet points here? This sentence is a little tricky to parse.

File: source/http-api.html.erb [hunk 2]
```diff
@@ -3,4 +3,74 @@ title: HTTP API
...
+
+    <h4>Java client examples</h4>
+    <p>TODO</p>
+  </section>
+
+    <section>
+    <h3>Dependency invocation endpoint</h3>
+
+    <p>
+      This is likely going to be your most-often used Octoparts endpoint. It is the endpoint that you will do a POST to in order to invoke your registered dependencies. In short, your POST body will be made of an AggregateRequest that contains a list of PartRequests, with each PartRequest corresponding to a single invocation of a registered dependency and containing all the necessary parameters required. The response from this endpoint will be an AggregateResponse, mainly consists of a series of PartResponses, each one corresponding to the response that each invoked registered dependency has sent in response to a PartRequest.
```
Comment (cb372) [inline source/http-api.html.erb]: typo: consisting

File: source/http-api.html.erb (hunk 1 above)
Comment (lloydmeta) [inline source/http-api.html.erb]: Good idea. Done.

File: source/http-api.html.erb (hunk 2 above)
Comment (lloydmeta) [inline source/http-api.html.erb]: Oops ! Fixed.

**Pull request merged**